service/k8sclient/k8client_test.go

```diff
@@ -36,75 +37,174 @@ func TestK8Client(t *testing.T) {
 	validKubeconfig, err := kubeCtl.Run(ctx, []string{"config", "view", "-o", "json"}, nil)
 	require.NoError(t, err)
 
-	client, err := NewK8Client(ctx, string(validKubeconfig))
+	client, err := New(ctx, string(validKubeconfig))
 	require.NoError(t, err)
 	t.Cleanup(func() {
 		err := client.Cleanup()
 		require.NoError(t, err)
 	})
 
-	name := "test-cluster"
-	_ = client.DeleteXtraDBCluster(ctx, name)
+	l := logger.Get(ctx)
 
-	for {
-		clusters, err := client.ListXtraDBClusters(ctx)
+	t.Run("XtraDB", func(t *testing.T) {
+		name := "test-cluster"
+		_ = client.DeleteXtraDBCluster(ctx, name)
+
+		for {
+			clusters, err := client.ListXtraDBClusters(ctx)
+			require.NoError(t, err)
+
+			l.Info(clusters)
+
+			if findXtraDBCluster(clusters, name) == nil {
+				break
+			}
+			time.Sleep(5 * time.Second)
+		}
+
+		err = client.CreateXtraDBCluster(ctx, &XtraDBParams{
+			Name: name,
+			Size: 2,
+		})
 		require.NoError(t, err)
+		for {
+			clusters, err := client.ListXtraDBClusters(ctx)
+			require.NoError(t, err)
 
-		if findCluster(clusters, name) == nil {
-			break
+			l.Info(clusters)
+
+			if cluster := findXtraDBCluster(clusters, name); cluster != nil && cluster.State == ClusterStateReady {
+				break
+			}
+			time.Sleep(5 * time.Second)
```

Comment:
Same comment as above. I'd move sleep before calling `clusters, err := client.ListXtraDBClusters(ctx)`. And in all similar loop.

BTW - I'd still add some timeout or number of tries in these "infinite" loops. I recently met an issue when minikube stuck - and did not respond for a long time - so I need to restart everything. So the timeout will be helpful, to at least ring a bell in a terminal.

Reply:
Timeout for request or for the test?

Reply:
> Timeout for request or for the test?

It's hard to answer right away. What will be easier to implement and support?

Reply:
Interesting thing here that minikube takes a lot of time to create cluster in CI

Reply:
Can we use EKS for CI?

Reply:
Good idea, we can try. Let's discuss with @AlekSi once he come back.